test/SILOptimizer/diagnostic_constant_propagation.swift

```diff
@@ -185,49 +185,53 @@ func testConvertOverflow() {
   Int64(UInt32(200)) // expected-warning{{unused}}
   UInt64(UInt32(200)) // expected-warning{{unused}}
 
-  // IEEE binary32 max value = 2^128 * (2^23-1)/2^23
-  var _ /*float32_max*/                     : Float32 = (340282326356119256160033759537265639424)
-  var _ /*float32_max_p1*/                  : Float32 = (340282326356119256160033759537265639425)
 
-  // 2^128 * (2^25-1)/2^25 - 1
-  var _ /*float32_max_not_yet_overflow*/    : Float32 = (340282356779733661637539395458142568447)
-  // 2^128 * (2^25-1)/2^25
+  var _ /*float32_max*/                     : Float32 = (340282346638528859811704183484516925440) // in hexfloat = 0x1.fffffep127
+
+  var _ /*float32_near_max_but_imprecise*/  : Float32 = (340282326356119256160033759537265639425) // expected-warning {{'340282326356119256160033759537265639425' is not exactly representable as 'Float32' (aka 'Float'); it becomes '3.40282326E+38'}}
```

Comment:
We probably want to be printing these with full precision, to make it clear what the difference is.

Reply:
@jrose-apple Yeah sure, let me make that change.